updates for issue#68 (#69)
SpencerDawkins/net-collab-rqmts: draft-kwbdgrr-tsvwg-net-collab-rqmts.md

~~~~~~~~
{: #Figure-netshaper title=”Metadata and Network Shaping”}

> {{Figure-netshaper}} shows a bottleneck (access) Router on the path of packets from Server to Client.

Reviewer: ```suggestion
{{Figure-netshaper}} shows a bottleneck (access) router on the path of packets from Server to Client.
```

> A network shaper in the Router manages QoS of flows of multiple users and can buffer (delay), discard or apply other flow control rules. Application layer signaling and feedback between client – server (A – in figure) adjusts rate over a period of several RTTs using feedback and congestion control algorithms.

Reviewer: ```suggestion
A network shaper in the router manages QoS of flows of multiple users and can buffer (delay), discard, or apply other flow control rules. Application layer signaling and feedback between Client – Server (A – in the figure) adjust transmission rate over a period of several RTTs using feedback and congestion control algorithms.
```

Congestion control algorithms are generally conservative and settle to a steady rate that avoids excessive packet loss.
In networks where link conditions (between Client and Router) vary significantly at timescales well below the RTT, this results in unused (wasted) bandwidth at short timescales.
There is some research {{5G-Octopus}} to indicate that media applications can obtain better QoE when sending at a higher rate (less conservative than current CCA) and the media application is willing to tolerate some packet loss or delay of low priority packets.
Packet priority and tolerance to delay of packets in such a case would be provided on-path in a side channel associated to the downstream packet (B – in figure).
The requirements for this server-to-network (S2N) metadata are described in {{server-network}}.

> Network shapers observe flows and apply policies to maximize performance but is not aware if there is a preference for one flow (UDP 4-tuple) over another flow belonging to the same user and network attachment (e.g., a subscriber connection, a 3GPP PDU session. See {{net-attach}} for more details).

Reviewer: ```suggestion
Network shapers observe flows and apply policies to maximize performance but are not aware whether  there is a high preference for one flow (UDP 4-tuple) over another flow belonging to the same user and network attachment (e.g., a subscriber connection, a 3GPP PDU Session. See {{net-attach}} for more details).
```

> The client can provide the (access) Router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in figure).

Reviewer: ```suggestion
Clients may provide an (access) router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in Figure).
```

> The client can provide the (access) Router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in figure).

Reviewer 2: Are we delving into inter-flow priorities, outside of the same UDP 4-tuple? I thought we mindfully wanted to move away due to a lot of complexities.

> The client may also provide information to the (access) Router to drop lower priority marked packets of a flow (UDP 4-tuple) temporarily which can in turn redirect bandwidth to other flows of that network attachment.

Reviewer: ```suggestion
Clients may also provide information to an (access) router to drop, when resources are congested,  'lower priority'-marked packets of a flow (UDP 4-tuple) temporarily which can in turn allocate bandwidth to other flows of that same network attachment.
```

# Use Cases {#uc}

> ## Streaming and Interactive Media {#uc-media}

Reviewer 2: IMHO, Even if the requirements are overlapping, Streaming and Interactive Media should be separate items, given they are 2 big use cases that comprise of most of the internet traffic. On the other hand, we can merge interactive media, accommodate delay and mixed traffic into one.

> Streaming media packets carry audio or video between server and client.

Reviewer: ```suggestion
Streaming media packets carry audio or video between servers and clients.
```

Examples include live broadcast and on-demand video streaming.

Interactive media includes content that a user can actively engage with and results in input and response actions that can be highly delay-sensitive.
Examples include VoIP (Peer-to-Peer (P2P), group conferencing), gaming and eXtended Reality (XR).

During congestion or other reactive policy events, retransmissions cause long delays.
All packets being treated the same can have challenges in efficiently handling/forwarding data.
Today, there is no way to identify packets which are less important and/or loss-tolerant to prioritize packets in challenging networks and/or during reactive events.
Some media frames may be able to tolerate more delay over the wire than others (e.g., live media frames require very low latency while a background image for augmented reality may be delivered with more delay tolerance).
Even when the media payload is not encrypted, the network has no means to distinguish these different requirements.

Requirements:

REQ-PACKET-PRIORITY: Packet priority relative to other packets in the transport flow (UDP 4-tuple).
REQ-PACKET-DELAY: Metadata to indicate whether the packet can tolerate delay.

> ## Mixed-Traffic and User Preferences {#uc-preferences}

Reviewer 2: User Preference might not belong here given that is agnostic of the traffic type but is an end user option. I am not sure if combining these 2 makes sense

> REQ-PACKET-PRIORITY: Packet priority relative to other packets in the transport flow (UDP 4-tuple).

Reviewer: ```suggestion
REQ-PACKET-PRIORITY: Packet priority relative to other packets within the same transport flow (UDP 4-tuple).
```

> REQ-PACKET-DELAY: Metadata to indicate whether the packet can tolerate delay.

Reviewer: ```suggestion
REQ-PACKET-DELAY: Metadata to indicate whether a packet can tolerate delay.
```

Some content is more resilient to buffering and delays (e.g., file copy, file download) compared to interactive traffic. This traffic can be buffered in favor of improved interactivity, without significant impact to the user experience.

> For example, an on-going file copy operation flow (UDP 4-tuple) a remote desktop user can tolerate more delay than the flow carrying graphics updates.

Reviewer: ```suggestion
For example, an on-going file copy operation flow (UDP 4-tuple) of a remote desktop user can tolerate more delay than the flow carrying graphics updates.
```

> Future versions of the document may re-consider this conclusion as a function of the comments received from the TSVWG.

Requirements: REQ-FLOW-PRIORITY in {{uc-preferences}} applies here.

> ### Delay Tolerance between Flows (Inter-flow) {#interflow-delay}

Reviewer 2: Again, we should refrain from any inter-flow discussions IMO.
